pyomo/solvers/plugins/solvers/CBCplugin.py

```diff
@@ -178,10 +189,98 @@ def _default_results_format(self, prob_format):
             return ResultsFormat.sol
         return ResultsFormat.soln
 
-    # Nothing needs to be done here
-    #def _presolve(self, *args, **kwds):
-    #    # let the base class handle any remaining keywords/actions.
-    #    SystemCallSolver._presolve(self, *args, **kwds)
+    def warm_start_capable(self):
+        if self._problem_format == ProblemFormat.cpxlp \
+                and _cbc_version >= (2,8,0,0):
+            return True
+        else:
+            return False
+
+    def _write_soln_file(self, instance, filename):
+
+        # Maybe this could be a useful method for any instance.
+
+        if isinstance(instance, IBlock):
+            smap = getattr(instance, "._symbol_maps")[self._smap_id]
+        else:
+            smap = instance.solutions.symbol_map[self._smap_id]
+        byObject = smap.byObject
+
+        with open(filename, 'w') as solnfile:
+            for var in instance.component_data_objects(Var):
+                if var.value and (id(var) in byObject):
```

Comment:
This appears to only write out nonzero values. Is that sufficient for CBC to accept a warmstart?

Reply:
Yes, in fact the CBC documentation for mipstart explicitly states that only non-zero integer/boolean variables are considered. I made some changes to clarify this and not write continuous variables either.